### PR TITLE
fix(site): show error on template upload failure

### LINKED ISSUE #15441: Web UI provides no feedback for a failed template upload
Image
Image

Additionally, clicking `Create Template` with this failed file upload also provides no feedback or evidence as to why the process failed.

This also means there's no feedback when uploading a file with a valid file extension, but a MIME type that's not exactly `application/x-tar` or `application/zip`, e.g. prior to #15442.

### PR DESCRIPTION
**Pull request by ethanndickson**
Closes #15441.

### CODE REVIEW

**Comment by ethanndickson** [comment]
* **#15558** <a href="https://app.graphite.dev/github/pr/coder/coder/15558?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/coder/coder/15558?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: site/src/pages/CreateTemplatePage/UploadTemplateView.tsx [hunk 1]
```diff
@@ -51,7 +53,14 @@ export const UploadTemplateView: FC<CreateTemplatePageViewProps> = ({
 			jobError={isJobError ? error.job.error : undefined}
 			logs={templateVersionLogsQuery.data}
 			upload={{
-				onUpload: uploadFileMutation.mutateAsync,
+				onUpload: async (file: File) => {
+					try {
+						await uploadFileMutation.mutateAsync(file);
+					} catch (error) {
```
Comment (ethanndickson) [inline site/src/pages/CreateTemplatePage/UploadTemplateView.tsx:59]: I'm not familiar with this query library, but what I found is that even though you can set `onError`, an exception returned from `mutateAsync` isn't caught unless you explicitly catch it. So, If I displayed the error as part of `onError` in `useMutation`, I'd still need to wrap `mutateAsync` in a try block, with an empty catch, which seems pointless.

Comment (BrunoQuaresma) [reply site/src/pages/CreateTemplatePage/UploadTemplateView.tsx:59]: You did it correctly. We usually use `mutateAsync` to keep the error handling close to where the mutation is called.

File: site/src/pages/CreateTemplatePage/CreateTemplatePage.test.tsx
Comment (BrunoQuaresma) [inline site/src/pages/CreateTemplatePage/CreateTemplatePage.test.tsx:1]: Nice tests 👍

**Review by BrunoQuaresma — APPROVED** [review]

**Pull request merged**